### ISSUE TITLE
Site editor: Navigator screens inconsistent icon buttons position and missing tooltip

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#52590]
### Description

Discovered while investigating https://github.com/WordPress/gutenberg/issues/52510

In the navigator screens, some editable objects (e.g. custom templates, custom patterns, pages) show the 'Actions' and 'Edit' icon buttons, in that order. However, for the Navigation menus these icon buttons are inconsistent:
- The order is reversed.
- The Actions icon button misses a tooltip.

<img width="750" alt="Screenshot 2023-07-13 at 09 17 34" src="https://github.com/WordPress/gutenberg/assets/1682452/3912f075-ab79-4c68-a7c3-5747fc76a4fb">

While the inconsistent order is a design / consistency / usability issue, the missing tooltip is an accessibility issue.

I'd like to remind that _any_ icon button _must_ visually expose its accessible name. Tooltips are required for these buttons. Since there have been several cases of icon buttons with no tooltip (and still there are a few to cover) I'd appreciate a bit more of attention to this requirement from everybody. If education and training turns out to be not enough to make sure icon buttons always have a tooltip, we should consider to make thie behavior automatic and / or implement tests for all icon buttons.

Since this seems a quick and self-contained fix, I'd suggest to consider it for inclusion in the approaching WP 6.3 release.

### Step-by-step reproduction instructions

- Go to the site editor.
- Observe the order of the buttons and whether they have a tooltip in the following screens:
  - Templates > A custom template.
  - Templates > A default template that hae been customized.
  - Pages > Any page
  - Navigation > Any menu

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
After more investigation, turns out the Actions button in the Navigation not only misses a tooltip: it misses a lable in the firrst place. One more case of unlabelled control.

**Comment by afercia** [comment]
Is there any way we can make the DropDownMenu `label` prop required? Cc @ciampo @mirka 
Introduced in https://github.com/WordPress/gutenberg/pull/50880

https://github.com/WordPress/gutenberg/blob/7d84f622a644cbcfb9111a288fa7efd0b83bad57/packages/edit-site/src/components/sidebar-navigation-screen-navigation-menu/more-menu.js#L34-L38
